### PR TITLE
Add article page and rendering download component

### PR DESCRIPTION
**Pull request by tomas-stefano**
[TP card](https://moneyadviceservice.tpondemand.com/entity/9161-fincap-article-page-view-or-add)

This commit adds the article page to the fincap app and also adds the download box rendering from CMS.

The server part of this work you can see [here](https://github.com/moneyadviceservice/cms/pull/428)

**Obs.:** Please review first https://github.com/moneyadviceservice/mas-cms-client/pull/24 !
So that I can change the Gemfile to point to a version instead of git

## Article components Pull requests

- [x] Download component (this PR)
- [x] CTA links component https://github.com/moneyadviceservice/fin_cap/pull/96
- [x] Feedback component https://github.com/moneyadviceservice/fin_cap/pull/97
- [ ] Hero image component

## Other part of the work

- [ ] CMS server part of the work (WIP PR) https://github.com/moneyadviceservice/cms/pull/428
- [x] Mas Cms client https://github.com/moneyadviceservice/mas-cms-client/pull/24

### CODE REVIEW

File: app/presenters/article_presenter.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class ArticlePresenter < BasePresenter
+  DOWNLOAD_COMPONENT_IDENTIFIER = 'component_download'.freeze
+
+  def download_component
+    view.render('components/download', content: download_content)
```
Comment (margOnline) [inline app/presenters/article_presenter.rb]: should this be a constant?

Comment (tomas-stefano) [reply app/presenters/article_presenter.rb]: I usually moved to a constant but I thought for a while is these are usually code from the view inside the presenter. It is just felt awkward to move to a constant a render partial_name.

**Comment by tomas-stefano** [comment]
Merged the PR but I am happy to do the fixes on #96

File: app/components/download_component.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class DownloadComponent
+  attr_reader :block, :document
+  HTML_ITEM_CLASS = 'download-box__list-item-link'.freeze
+
+  def initialize(block)
+    @block = block
+  end
+
+  def process
```
[the remark this answers is not in the document]
Comment (willhall88) [reply app/components/download_component.rb]: @tomas-stefano and I chatted about this offline and bounced around a lot of ideas on what this should be. We're thinking something along the lines of `build_markup` .. wdyt?

File: app/views/articles/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<div class="l-constrained">
+  <div class="hero">
+    <h1 class="hero__heading">
+      HERO component
```
Comment (willhall88) [inline app/views/articles/show.html.erb]: I'm assuming this text is going to change in the future?

Comment (tomas-stefano) [reply app/views/articles/show.html.erb]: Yes! It will change.

**Pull request merged**